jupyter-extension/src/plugins/mount/components/Pipeline/hooks/usePipeline.tsx

```diff
@@ -41,68 +44,54 @@ export const usePipeline = (
     }
   }, [metadata]);
 
+  let input;
+  try {
+    input = YAML.parse(inputSpec);
+  } catch (e) {
+    if (e instanceof YAML.YAMLParseError) {
+      input = JSON.parse(inputSpec);
+    } else {
+      throw e;
+    }
+  }
+
+  const samemeta: SameMetadata = {
+    apiVersion: 'sameproject.ml/v1alpha1',
+    environments: {
+      default: {
+        image_tag: imageName,
+      },
+    },
+    metadata: {
+      name: pipelineName,
+      version: '0.0.0',
+    },
+    notebook: {
+      requirements: requirements,
+    },
+    run: {
+      name: pipelineName,
+      input: JSON.stringify(input),
+    },
+  };
+
```

Comment:
I moved these out of the functions to avoid repeated code. I _think_ this is OK because any changes to these values cause a re-render of the Pipeline extension. @pleeko could you do a quick sanity check?

Reply:
Huh interesting, I confirmed `usePipeline` is getting called every time the metadata changes, but I'm curious if we actually want this behavior. Just to check my understanding: The `metadata` prop is used to set the metadata on initial component load, but from then on, the `set` state methods and the `useEffect` hook are used to track component state, and the component shouldn't get re-rendered for these internal state changes.